src/parquet/encodings/dictionary-encoding.h

```diff
@@ -418,6 +432,23 @@ inline void DictEncoder<FixedLenByteArray>::WriteDict(uint8_t* buffer) {
   }
 }
 
+template<typename T>
+inline void DictEncoder<T>::FreeMemory() {}
+
+template<>
+inline void DictEncoder<ByteArray>::FreeMemory() {
+  for (const auto& value : uniques_) {
+    pool_->Free(const_cast<uint8_t*>(value.ptr));
+  }
+}
+
+template<>
+inline void DictEncoder<FixedLenByteArray>::FreeMemory() {
+  for (const auto& value : uniques_) {
+    pool_->Free(const_cast<uint8_t*>(value.ptr));
+  }
+}
```

Comment:
I would suggest that `ChunkedMemoryPool` not be a subclass of `MemoryPool` and instead take a memory pool as a parameter. That way we won't have this problem. The MemPool from Impala only really makes sense with a bunch of allocations followed by `FreeAll` (in the context of a step in a data flow system / SQL engine)